test/penalizer/Penalizer.test.ts

```diff
@@ -290,6 +273,81 @@ contract('Penalizer', function ([relayOwner, relayWorker, relayManager, otherAcc
     })
   })
 
+  describe('should receive/reject claims according with the commitment time', function () {
+    before(async function () {
+      await penalizer.setHub(relayHub.address, { from: await penalizer.owner() })
+    })
+    beforeEach(async () => {
+      await token.mint('1000', forwarder.address)
+
+      await relayHub.stakeForAddress(relayManager, 1000, {
+        from: relayOwner,
+        value: ether('1'),
+        gasPrice: gasPrice
+      })
+      await fundAccount(fundedAccount, sender.address, '10')
+    })
+
+    // successful claims
+    const acceptableCommitmentTimes = [
+      // due to security implications, we accept a up to 15 secs of delay
+      // but we cannot set exactly this time in tests
```

Comment:
i've been thinking about this:

granted, even when using `now` and things like `sleep` statements, we cannot guarantee the _exact_ time of execution for any given line, and i suppose there is even more uncertainty when block timestamps are involved.

the question is: do we need to? or do we only need to make sure a given time has elapsed? (or make sure that it has not)

i think that by using small or large enough values, we can achieve this with a very good degree of certainty. 

do you mean that we cannot test an edge case of, say, presenting a claim within 1 second of it being expired? 

in that case that's fine, i think we can accept this as an expected limitation

Reply:
>do you mean that we cannot test an edge case of, say, presenting a claim within 1 second of it being expired?

yep, I meant exactly that, and I have already tried to set values as close as possible, without impacting the developer experience (e.g. I don't want failing tests because of the time)

Reply:
then we agree. resolving this